### PR TITLE
Unstructured api init

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

@natsukium sorry to ask for your help always!
I've got a python package that doesn't have a setup phase, but has dependencies in a requirement.txt
https://github.com/Unstructured-IO/unstructured-api/blob/main/requirements/base.in
the start script is just calling uvicorn though
https://github.com/Unstructured-IO/unstructured-api/blob/main/scripts/app-start.sh
I was wondering if using makeWrapper is the right way here ?

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
The last error I get is `ModuleNotFoundError: No module named 'prepline_general'`
most likely the way I'm copying the python files is wrong.

**Comment by happysalada** [comment]
This looks like it's ready, the web server starts
(you can pass --host or --port to change the default host or port).
The NLTK_DATA trick looks like it's working.
I'm going to wait a couple more days on this, but this is ready to merge IMO.

File: pkgs/servers/unstructured-api/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  };
+in stdenvNoCC.mkDerivation {
+  pname = "unstructured-api";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "Unstructured-IO";
+    repo = "unstructured-api";
+    rev = version;
+    sha256 = "sha256-fk0YkGllggi0eWdp9ytHy4/9rChkcDnQvEvVAp1+RJw=";
```
Comment (natsukium) [inline pkgs/servers/unstructured-api/default.nix:38]:
```suggestion
    hash = "sha256-fk0YkGllggi0eWdp9ytHy4/9rChkcDnQvEvVAp1+RJw=";
```

File: pkgs/servers/unstructured-api/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```
Comment (natsukium) [inline pkgs/servers/unstructured-api/default.nix:43]:
```suggestion
  installPhase = ''
    runHook preInstall

```

File: pkgs/servers/unstructured-api/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    mkdir -p $out $out/bin $out/lib
+    cp -r . $out/lib
+
+    makeWrapper ${pythonEnv}/bin/uvicorn $out/bin/unstructured-api \
+      --set NLTK_DATA ${unstructured_api_nltk_data} \
+      --prefix PYTHONPATH : $out/lib \
+      --add-flags "prepline_general.api.app:app"
```
Comment (natsukium) [inline pkgs/servers/unstructured-api/default.nix:50]:
```suggestion
      --add-flags "prepline_general.api.app:app"

  runHook postInstall
```

**Review by natsukium — COMMENTED** [review]
only nits

**Comment by happysalada** [comment]
Thanks for coming by!
All should be fixed.

**Review by natsukium — APPROVED** [review]
It works fine for me.
I tried these examples.
https://github.com/Unstructured-IO/unstructured-api/tree/0.0.39#using-the-api-locally

**Pull request merged**